R/settings.R

```diff
@@ -98,7 +98,10 @@ read_config_file <- function(config_file) {
   if (endsWith(config_file, ".R")) {
     load_config <- function(file) sys.source(file, config, keep.source = FALSE, keep.parse.data = FALSE)
     malformed <- function(e) {
-      stop("Malformed config file, ensure it is valid R syntax\n  ", conditionMessage(e), call. = FALSE)
+      cli_abort(c(
+        "Malformed config file, ensure it is valid R syntax.",
+        conditionMessage(e)
```

Comment:
Should we use `i = ` here as well as below?

Reply:
The problem is that when the condition message (in `e`) itself is formatted with `{cli}`, either the formatting looks messed up, or part of the condition message is dropped.

E.g. notice that part of the condition message is dropped here:

``` r
foo <- function(expr) {
  cli::cli_abort(c(
    i = "{.arg expr} must be an {.cls xml_nodeset} or an {.cls xml_node}.",
    x = "Instead, it is {.obj_type_friendly {expr}}."
  ))
}

tryCatch(
  foo(1),
  error = function(e) {
    print(e)
    cat("---------------------------\n")
    cli::cli_inform(c(
      i = "Something went wrong here.",
      x = e$message
    ))
  }
)
#> <error/rlang_error>
#> Error in `foo()`:
#> ℹ `expr` must be an <xml_nodeset> or an <xml_node>.
#> ✖ Instead, it is a number.
#> ---
#> Backtrace:
#>     ▆
#>  1. ├─base::tryCatch(...)
#>  2. │ └─base (local) tryCatchList(expr, classes, parentenv, handlers)
#>  3. │   └─base (local) tryCatchOne(expr, names, parentenv, handlers[[1L]])
#>  4. │     └─base (local) doTryCatch(return(expr), name, parentenv, handler)
#>  5. └─global foo(1)
#> ---------------------------
#> ℹ Something went wrong here.
#> `expr` must be an <xml_nodeset> or an <xml_node>.
```

<sup>Created on 2024-05-29 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

@olivroy Do you know how to deal with this?

Reply:
I think it is better to use the following:

```r
cli::cli_abort(
  "message",
  parent = e
)
```

If you want me to do it, you can leave it for now, open a new issue and I can open a PR to address that post-merge!

Reply:
Thanks! I think, for now, I am going to leave this message as is.

I have created a new issue for a follow-up: https://github.com/r-lib/lintr/issues/2588. We can work on that once the current PR is merged.